src/hooks/useConcreteNotification.ts

```diff
@@ -15,22 +16,34 @@ const concreteNotificationQuery = gql`
   }
 `
 
-const useConcreteNotification = (id: number) => {
-  const { data, loading, error, refetch } = useQuery(
+const isMessageValid = (message: UserNotification | null): boolean => {
+  if (message === null) return false
+
+  const requiredFields = ["id", "headline", "body", "notification"]
+  const fields = Object.keys(message)
+
+  for (const requiredField of requiredFields) {
+    if (!fields.includes(requiredField)) return false
+  }
+
+  return true
+}
+
+export const useConcreteNotification = (id: number | null) => {
+  const { data, loading, error } = useQuery(
     concreteNotificationQuery,
     {
-      variables: { id }
+      variables: { id },
+      skip: id == null
```

Comment:
```suggestion
      skip: !id
```
This should handle undefined, null & empty string. 🙂